src/hdmf_zarr/backend.py

```diff
@@ -426,6 +450,50 @@ def write_builder(self, **kwargs):
         self.logger.debug("Done writing %s '%s' to path '%s'" %
                           (f_builder.__class__.__qualname__, f_builder.name, self.source))
 
+        # Consolidate metadata for the entire file after everything has been written
+        if consolidate_metadata:
+            zarr.consolidate_metadata(store=self.path)
+
+    @staticmethod
+    def __get_store_path(store):
+        """
+        Method to retrieve the path from the Zarr storage.
+        ConsolidatedMetadataStore wraps around other Zarr Store objects, requiring a check to
+        retrieve the path.
+        """
+        if isinstance(store, zarr.storage.ConsolidatedMetadataStore):
+            fpath = store.store.path
+        else:
+            fpath = store.path
+
+        return fpath
+
+    def __open_file_consolidated(self,
+                                 store,
+                                 mode,
+                                 synchronizer=None,
+                                 storage_options=None):
+        """
+        This method will check to see if the metadata has been consolidated.
+        If so, use open_consolidated.
+        """
+        # self.path can be both a string or a one of the `SUPPORTED_ZARR_STORES`.
+        if isinstance(self.path, str):
+            path = self.path
+        else:
+            path = self.path.path
+
+        if os.path.isfile(path+'/.zmetadata'):
```

Comment:
```suggestion
        if os.path.isfile(path+'/.zmetadata') and mode not in ['r', 'r+']:
```

Do we also need to check for the supported mode here?

Reply:
No. There's only two places it's used. One has the check (in open) and the other is in read mode.